image/scanner/konflux/Dockerfile

```diff
@@ -0,0 +1,62 @@
+ARG BASE_REGISTRY=registry.access.redhat.com
+ARG BASE_IMAGE=ubi8-minimal
+ARG BASE_TAG=latest
+
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_1.20 as builder
+
+ENV CGO_ENABLED=1
+ENV GOFLAGS=""
+ENV CI=1
+
+COPY . /src
+WORKDIR /src
+
+RUN unzip -j blob-repo2cpe.zip -d image/scanner/dump/repo2cpe && \
+    unzip -j blob-k8s-definitions.zip -d image/scanner/dump/k8s_definitions && \
+    unzip -j blob-nvd-definitions.zip -d image/scanner/dump/nvd_definitions
+
+RUN echo "version: $(git describe --tags --abbrev=10 --dirty --long)"; git status; go build -trimpath -ldflags="-X github.com/stackrox/scanner/pkg/version.Version=$(git describe --tags --abbrev=10 --dirty --long)" -o image/scanner/bin/scanner ./cmd/clair
+
+# Replace genesis manifests file in the source code with the one generated at
+# the point when the dump was taken.  This is to avoid discrepancy between other
+# files of the dump and the manifest.
+COPY ./blob-genesis_manifests.json image/scanner/dump/genesis_manifests.json
+
+FROM ${BASE_REGISTRY}/${BASE_IMAGE}:${BASE_TAG}
+
+LABEL name="scanner" \
+      vendor="StackRox" \
+      maintainer="support@stackrox.com" \
+      summary="Image scanner for the StackRox Kubernetes Security Platform" \
+      description="This image supports image scanning in the StackRox Kubernetes Security Platform."
+
+ENV NVD_DEFINITIONS_DIR="/nvd_definitions"
+ENV K8S_DEFINITIONS_DIR="/k8s_definitions"
+ENV REPO_TO_CPE_DIR="/repo2cpe"
+
+COPY --from=builder /src/image/scanner/scripts /
+COPY --from=builder /src/image/scanner/bin/scanner ./
+COPY --chown=65534:65534 --from=builder "/src/image/scanner/dump${NVD_DEFINITIONS_DIR}/" ".${NVD_DEFINITIONS_DIR}/"
+COPY --chown=65534:65534 --from=builder "/src/image/scanner/dump${K8S_DEFINITIONS_DIR}/" ".${K8S_DEFINITIONS_DIR}/"
+COPY --chown=65534:65534 --from=builder "/src/image/scanner/dump${REPO_TO_CPE_DIR}/" ".${REPO_TO_CPE_DIR}/"
+COPY --chown=65534:65534 --from=builder /src/image/scanner/dump/genesis_manifests.json ./
+
+RUN microdnf upgrade --nobest && \
+    microdnf install xz && \
+    microdnf clean all && \
+    # (Optional) Remove line below to keep package management utilities
+    # We don't uninstall rpm because scanner uses it to get packages installed in scanned images.
+    rpm -e --nodeps $(rpm -qa curl '*dnf*' '*libsolv*' '*hawkey*' 'yum*') && \
+    rm -rf /var/cache/dnf /var/cache/yum && \
+    chown -R 65534:65534 /tmp && \
```

Comment:
In addition to `/tmp`, the downstream dockerfile seems to change ownership of a few other things.

```bash
    chown -R 65534:65534 \
        genesis_manifests.json \
        "${NVD_DEFINITIONS_DIR}" \
        "${K8S_DEFINITIONS_DIR}" \
        "${REPO_TO_CPE_DIR}" \
        /tmp
```

I like none of that and I also found it difficult to find out why this is done for some similar things in the past.

This comment is not blocking for your PR but if you want to cargo-cult additional locations in the `chown` command, this _might_ save us from some issues during later testing.

Reply:
Those extra files are `chown`'d in their [respective `COPY` lines](https://github.com/stackrox/scanner/blob/60685202bfdde41fb73d2d2fd90b06d75a005f28/image/scanner/konflux/Dockerfile#L53-L56).  According to https://issues.redhat.com/browse/RS-260, [scanner runs in the container as the nobody user](https://github.com/stackrox/stackrox/blob/e0911d2e77f8c3f0868cf45f267f3f0cead93a8c/image/templates/helm/shared/templates/02-scanner-06-deployment.yaml.htpl#L88).

Reply:
Ah, that makes sense: I did not pay attention that there's `COPY` with `--chown`.

What bugs me about the ownership change: the container is supposed to run on the cluster with read-only root filesystem so these files are not supposed to be modified. What that means to me, the desire is to open the files for _reading_ by the executing user (uid=65534). In such case, giving ownership is overly permissive, it could be sufficient to do `chown`. It could be tightened up with some regression testing but it's certainly not in scope of this work here, we could just make sure we don't make it worse.